### DISCUSSION TITLE
problem with source 96188 from Cesium Ion

### DISCUSSION THREAD

**Discussion by pcardinal** [iTowns/itowns#2071] [General]
With the following code :
```
            // Create a 3D Tiles layer from Cesium ion server with Cesium.
            var threeDTilesIonSource = new itowns.C3DTilesIonSource({
                accessToken: 'eyJhbGciOiJIUzI1NiIsInR5.....',  
                assetId: 96188
            });
            threeDTilesIonSource.whenReady.then(displayAttributions); 
            var threeDTilesIonLayer = new itowns.C3DTilesLayer('3d-tiles-cesium-ion', {
                name: '3D Tiles from Cesium Ion',
                source: threeDTilesIonSource,
                registeredExtensions: extensions,
            }, view);
```
(The problem is not with the accessToken)

I got the following message in Chrome

```
Uncaught (in promise) RangeError: Invalid typed array length: 67884
    at new Float32Array (<anonymous>)
    at PntsParser.js:92:25
    at PntsParser.js:55:17
    at Z (3dTilesProvider.js:31:10)
    at 3dTilesProvider.js:119:18

```
In Firefox, I got the following message
```
Uncaught (in promise) RangeError: attempting to construct out-of-bounds Float32Array on ArrayBuffer
    o PntsParser.js:92
    Z PntsParser.js:55
    Z 3dTilesProvider.js:31
    executeCommand 3dTilesProvider.js:119

```

Is it possible to fix the RangeError problem ? I am searching in the code and I don't find where is the problem ?

 The assetId 96188 is a 3D Tiles (pnts) from a 10 billions cloud points (City of Montreal).

I don't have problems to display other 3DTiles from Cesium Ion (smaller files)

**Comment by jailln** [comment]
Hi,
Thanks for reporting this issue! It happens when point clouds have the [3DTILES_draco_point_compression](https://github.com/CesiumGS/3d-tiles/tree/main/extensions/3DTILES_draco_point_compression) extension that is not yet supported by itowns, see #1992.
Let me know if you're willing to implement it I can help you with the current code.

I will close this discussion, let's continue in #1992 if needed.
